substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/jfr/remote/JfrAutoSessionManager.java

```diff
@@ -0,0 +1,152 @@
+package com.oracle.svm.core.jdk.jfr.remote;
+
+import com.oracle.svm.core.jdk.jfr.JfrOptions;
+import com.oracle.svm.core.jdk.jfr.logging.JfrLogger;
+
+import jdk.jfr.Configuration;
+import jdk.jfr.Recording;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.text.ParseException;
+
+/*
+ * Handlers the startup and teardown of JFR when run from either the command line or remote interface.
+ * Implemented by standard JFR calls as user code might do.
+ */
+
+public class JfrAutoSessionManager {
+
+    private static final String DEFAULT_JFR_FILENAME = "recording.jfr";
+    private static final String DEFAULT_JFR_CONFIG = "default";
+
+    Configuration jfrConfiguration;
```

Comment:
Can these be made private? And for jfrConfiguration, final?

Reply:
done.